File: skins/MinervaNeue/tests/browser/features/step_definitions/notification_steps.rb
When /^I click on the notification icon$/ do
  on(ArticlePage) do |page|
    page.wait_until_rl_module_ready('skins.minerva.notifications')
    page.notifications_button_element.when_present.click
  end
end

Given(/^I have no notifications$/) do
  expect(on(ArticlePage).notifications_button_element.when_present).to be_visible
  # This is somewhat hacky, but I don't want this test making use of Echo's APIs which may change
  browser.execute_script("$( function () { $( '.notification-count span' ).hide(); } );")
end

When(/^I click the notifications overlay close button$/) do
  sleep 1
  on(ArticlePage).notifications_overlay_close_button_element.when_present.click
end

When(/^the notifications overlay appears$/) do
  on(ArticlePage) do |page|
    page.wait_until_rl_module_ready('oojs-ui-core')
    page.notifications_overlay_element.when_present
  end
end

Then(/^after (.+) seconds I should not see the notifications overlay$/) do |seconds|
  sleep seconds.to_i
  expect(on(ArticlePage).notifications_overlay_element).not_to be_visible
end

Then(/^I should see the notifications overlay$/) do
  expect(on(ArticlePage).notifications_overlay_element.when_present).to be_visible
end

File: skins/MinervaNeue/tests/selenium/features/step_definitions/notification_steps.js
'use strict';

const ArticlePage = require( '../support/pages/article_page' );
const Util = require( 'wdio-mediawiki/Util' );
const { iClickTheOverlayCloseButton } = require( './common_steps' );

const iHaveNoNotifications = async () => {
	await ArticlePage.notifications_button_element.waitForDisplayed();
	// This is somewhat hacky, but we don't want this test making use of
	// Echo's APIs which may change
	await browser.execute( '$( () => { $( ".notification-count span" ).hide(); } );' );
};

const iClickOnTheNotificationIcon = async () => {
	await Util.waitForModuleState( 'skins.minerva.scripts' );
	await ArticlePage.notifications_button_element.waitForDisplayed();
	await ArticlePage.notifications_button_element.click();
};

const iClickTheNotificationsOverlayCloseButton = async () => {
	await iClickTheOverlayCloseButton();
};

module.exports = {
	iHaveNoNotifications, iClickOnTheNotificationIcon,
	iClickTheNotificationsOverlayCloseButton
};
